civiform/civiform
# Make close button on modals accessible.

### Description

A few pieces in this PR:
1) Updates the close button icon to be inside a `<button>` with a valid aria-label.
2) Since the aria-label needs to be translated for applicant-facing modals, updates the `Modal.Builder` to take in (1) A `Location`, which can be `ADMIN_FACING`, `APPLICANT_FACING`, or `DEBUG`; and (2) An optional `Messages` object that has the translated messages. If the `Location` is `APPLICANT_FACING`, then the modal is *required* to set the `Messages` object so that the close button aria label can be correctly translated.
3) Moves the close button to be at the end of the DOM instead of in the header, per https://designsystem.digital.gov/components/modal/.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings) -- **Will wait until translations for "Close" have merged before merging this PR**
- [x] Manually evaluated tab order -- **Specifically verified the close button is last in the tab order within the modal.**

### Instructions for manual testing

1) Start a new session on CiviForm and start an application. Verify that the close button on the "Create an account" modal can be tabbed to, has a description read out by screen reader, and that pressing enter activates the button and closes the dialog. 
2) Verify the close button description changes based on the selected language.
3) As an admin, try adding questions to a program. Verify the items listed in step 1 but with the close button on the question bank. 

### Issue(s) this completes

Fixes #5520; Fixes #5521
